lib/system/external.toit

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import io
+import rpc
+
+clients_ ::= {:}
+notification-listeners_ := 0
+notification-handler_/ExternalMessageHandler_? := null
+
+class Client:
+  pid/int
+  id/string
+  notification-callback_/Lambda? := null
+  is-closed_/bool := false
+
+  constructor.private_ .pid .id .notification-callback_:
+
+  static open id/string --notification-callback/Lambda?=null -> Client?:
```

Comment:
Maybe we should use `on-notify` or `on-message` rather than `notification-callback`?